src/components/icon/icon.tsx

```diff
@@ -146,11 +151,11 @@ export class Icon {
   }
 
   render() {
-    const { iconName, el, inheritedAttributes } = this;
+    const { inheritedAttributes } = this;
     const mode = this.mode || 'md';
-    const flipRtl =
-      this.flipRtl ||
-      (iconName && (iconName.indexOf('arrow') > -1 || iconName.indexOf('chevron') > -1) && this.flipRtl !== false);
+    this.shouldAutoFlip =
+      (this.iconName?.includes('arrow') || this.iconName?.includes('chevron')) && this.flipRtl !== false;
+    this.shouldBeFlippable = this.flipRtl || this.shouldAutoFlip || this.el.hasAttribute('dir');
```

Comment:
```suggestion
    const shouldAutoFlip = (iconName && (iconName.indexOf('arrow') > -1 || iconName.indexOf('chevron') > -1) && flipRtl !== false);
    const shouldBeFlippable = flipRtl || shouldAutoFlip || this.el.hasAttribute('dir');
```

A few things:

1. IMO if we are always computing these values on `render` then there does not seem to be much value in caching them as global variables within the class. A local `const` should be fine here.
2. Using optional chaining has caused issues in the past with boolean/numeric values, so let's keep the old syntax just to be safe. (See https://github.com/ionic-team/ionic-framework/pull/25102 for an example)
3. Do we need `this.el.hasAttribute('dir')`? I believe `:dir()` should account for this. Or is this for browsers that support neither :host-context or :dir?

Reply:
Oh yeah, good call on `this.el.hasAttribute('dir')`. I think that was left over from when there were observers.